aspnetcore/blazor/file-uploads.md

```diff
@@ -101,26 +101,88 @@ If you're using the [Autofac Inversion of Control (IoC) container](https://autof
 
 ## File size read and upload limits
 
-:::moniker range=">= aspnetcore-6.0"
+:::moniker range=">= aspnetcore-9.0"
+
+Server-side or client-side, there's no file read or upload size limit for the <xref:Microsoft.AspNetCore.Components.Forms.InputFile> component.
+
+For Chromium-based browsers (for example, Google Chrome and Microsoft Edge) using the HTTP/2 protocol, HTTPS, and [CORS](xref:security/cors), client-side Blazor uses [Streams API](https://developer.mozilla.org/docs/Web/API/Streams_API) to permit uploading large files (over 250 MB or larger than the device's available memory) with [request streaming](xref:blazor/call-web-api#client-side-request-streaming).
+
+For non-Chromium browsers or without HTTP/2 protocol/HTTPS/CORS, client-side Blazor reads the file's bytes into a single JavaScript array buffer when marshalling the data from JavaScript to C#, which is limited to 2 GB or to the device's available memory. Large file uploads, typically 250 MB or larger, may fail for client-side uploads using the <xref:Microsoft.AspNetCore.Components.Forms.InputFile> component. For more information, see the following discussions:
+
+* [The Blazor InputFile Component should handle chunking when the file is uploaded (dotnet/runtime #84685)](https://github.com/dotnet/runtime/issues/84685)
+* [Request Streaming upload via http handler (dotnet/runtime #36634)](https://github.com/dotnet/runtime/issues/36634)
+
+For large client-side file uploads that don't meet the preceding criteria for request streaming and fail when attempting to use the <xref:Microsoft.AspNetCore.Components.Forms.InputFile> component, we recommend chunking large files with a custom component using multiple [HTTP range requests](https://developer.mozilla.org/docs/Web/HTTP/Range_requests) instead of using the <xref:Microsoft.AspNetCore.Components.Forms.InputFile> component.
```

Comment:
@javiercn Can you please confirm that we want to formalize the HTTP range request recommendation? Unless we've proven it's viable with a sample implementation, I'd prefer just to leave it out.

Reply:
References to HTTP range requests ... ***Removed!*** 🔪 on the latest updates.

Reply:
I spoke to @javiercn about this and he is fine with this removal.